libraries/AP_NavEKF3/AP_NavEKF3_OptFlowFusion.cpp

```diff
@@ -289,8 +289,12 @@ void NavEKF3_core::FuseOptFlow(const of_elements &ofDataDelayed, bool really_fus
     ftype vd = stateStruct.velocity.z;
     ftype pd = stateStruct.position.z;
 
-    // constrain height above ground to be above range measured on ground
+    // constrain height above ground to be above range measured on ground, or user specified height (if given, for rover)
+#if APM_BUILD_TYPE(APM_BUILD_Rover)
```

Comment:
@priseborough why is this working? @stephendade reports he can get optical flow fusion with no rangefinder. Does this mean the EKF is not properly checking the health of the rangefinder before fusing flow data?

Reply:
The optical flow fusion shouldn't be checking the range finder, it should be checking that the terrain height offsets state estimate is still valid. I had a quick look at the code and it does not appear that this is being done.